src/plugins/intel_cpu/src/nodes/mkldnn_reduce_node.cpp

```diff
@@ -1848,20 +1890,44 @@ void MKLDNNReduceNode::prepareParams() {
         set_reduce_dim_flags();
     }
 
-    if (compile_post_kernel) {
-        setPostOps(attr, dst_dims, true);
+    auto builder = [&](const ReduceKey& key) -> std::shared_ptr<jit_uni_reduce_post_kernel> {
+        std::shared_ptr<jit_uni_reduce_post_kernel> post_kernel;
+        auto post_jcp = jit_reduce_config_params();
+        post_jcp.layout = key.layout;
+        post_jcp.reduce_mode = key.reduce_mode;
+        post_jcp.src_dt = key.src_dt;
+        post_jcp.dst_dt = key.dst_dt;
+        post_jcp.src_data_size = MKLDNNExtensionUtils::sizeOfDataType(post_jcp.src_dt);
+        post_jcp.dst_data_size = MKLDNNExtensionUtils::sizeOfDataType(post_jcp.dst_dt);
+
         if (mayiuse(cpu::x64::avx512_common)) {
-            reduce_post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::avx512_common>(jcp, *attr.get()));
+            post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::avx512_common>(post_jcp, *attr.get()));
         } else if (mayiuse(cpu::x64::avx2)) {
-            reduce_post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::avx2>(jcp, *attr.get()));
+            post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::avx2>(post_jcp, *attr.get()));
         } else if (mayiuse(cpu::x64::sse41)) {
-            reduce_post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::sse41>(jcp, *attr.get()));
+            post_kernel.reset(new jit_uni_reduce_post_kernel_f32<cpu::x64::sse41>(post_jcp, *attr.get()));
         }
-        if (reduce_post_kernel)
-            reduce_post_kernel->create_ker();
+        if (post_kernel)
+            post_kernel->create_ker();
+
+        return post_kernel;
+    };
+
+    if (compile_post_kernel) {
+        setPostOps(attr, dst_dims, true);
+
+        auto selected_pd = getSelectedPrimitiveDescriptor();
+        ReduceKey key = {jcp.layout, jcp.reduce_mode, selected_pd->getImplementationType(), jcp.src_dt, jcp.dst_dt, attr.get_post_ops()};
+        auto cache = getRuntimeCache();
+        auto result = cache->getOrCreate(key, builder);
+        if (!result.first) {
+            IE_THROW() << "jit_uni_reduce_post_kernel_f32 was not found for node " << getName() << ".";
+        }
+
+        reduce_post_kernel = result.first;
         jit_mode = jit_mode && reduce_post_kernel;
 
-        if (!isDynamicNode() || (isDynamicNode() && attr.get()->post_ops_.len() == 0)) {
+        if (!isDynamicNode()) {
```

Comment:
Why this condition was changed?

Reply:
I think this is a petential bug. When this "if" condition meets, the reduce_post_kernel will never be recompiled again. But for dynamic reduce node, post_ops_ empty once doesn't make it empty for ever. If  post_ops_.len() changes later, the reduce_post_kernel should have the chance to be recompiled again. Further discussion is always welcomed. Thanks Dmitry!